tests/test_bs.py

```diff
@@ -230,11 +236,23 @@ def _setup_clients_in_namespace(name_space):
                                  container_specs=cspec)
 
         client_info.deployment_name = resp.metadata._name
-        namespaced_pods = client.CoreV1Api().list_namespaced_pod(namespace=name_space, include_uninitialized=True).items
-        client_pods = list(filter(lambda i: i.metadata.name.startswith(client_info.deployment_name), namespaced_pods))
+        client_pods = (
+            client.CoreV1Api().list_namespaced_pod(name_space,
+                                                   include_uninitialized=True,
+                                                   label_selector=(
+                                                       "name={0}".format(
+                                                           client_info.deployment_name.split('-')[0]))).items)
 
         client_info.pods = [{'name': c.metadata.name, 'pod_ip': c.status.pod_ip} for c in client_pods]
         return client_info
+
+    def fin():
+        # On teardown we wish to report on pods that were restarted by k8s during the test
+        restarted_pods = pod.check_for_restarted_pods(testconfig['namespace'])
```

Comment:
this will definitely work here, but I think it is more suitable in a more central function like the namespace creation (and deletion).
as this checks for pods in the whole namespace and not not just the clients.

Reply:
The check should be in the first teardown fixture called.
Otherwise, we might miss some pods which may be deleted in other teardown fixtures.
The alternative was to search for pods for the current deployment (rather than the whole namespace) but here we might end up with multiple API calls and impact on the run time

Reply:
currently we only teardown pods by tearing down the namespace. hence that wouldn't happen in the current code.

Reply:
you don't really know that this is the first teardown fixture called anyway. I thought this is a result of the order of the fixtures.